web/api/web_api_v1.c

```diff
@@ -212,6 +212,43 @@ inline int web_client_api_request_v1_alarms(RRDHOST *host, struct web_client *w,
     return 200;
 }
 
+#define CHECK_HTTP400(x) { if(!(x)) { error("CHECK_HTTP400(%s)", #x); return 400; } }
+
+inline int web_client_api_request_v1_alarm_counts(RRDHOST *host, struct web_client *w, char *url) {
+    int i = 0;
+
+    CHECK_HTTP400(host);
```

Comment:
This check and the other two are safety, without any doubt, but "host" and the "w" will never be NULL here, case you follow the whole connection process in the files libnetdata/socket/socket.c and web/server/static/static-threaded.c you will see that these structures are created in the first step of the connection, and case there is a problem, the code won't go in front. In my vision they are not necessary.

Reply:
These checks are for safety in case of code refactoring, i.e. the function should not crash even something will be changed outside of it in future. I do not want to advertise these checks too much, but other software like augeas has similar checks and I though they will be usefull for netdata. Especially because netdata is a server application.

If you think checks are redundant here then I will remove them.

Reply:
Hi @alpes214,

I understand that this append a safety level for the function, but I will explain more the whole situation and you will see that for the specified variables  it is impossible to arrive in this level with NULL values.

| accep_socket | libnetdata/socket/soscket.c | accepts the connection
| poll_add_fd | libnetdata/socket/soscket.c | add it to the pool of connections
| web_server_add_callback | web/server/static/static-threaded.c | Add callbacks to the connection
| web_client_create_on_fd | web/server/static/static-threaded.c | Call the function to create web_client

I am jumping some steps, because it is not the focus of this PR.

| web_server_rcv_callback | web/server/static/static-threaded.c | Call the function web_client_process_request
| web_client_process_request | web/server/web_client.c | call the function  web_client_process_url

The arguments of this function has localhost that is created in the start of the netdata(database/rrdhost.c), you have the "w" that was the web client structure initialized in the previous functions and the url_decoded that is filled inside the web_client.c

When you said "refactoring code", for this specific case, it is refactoring the whole Netdata, so we do not need to worry with this, because this will not be necessary.

Another important point, the Netdata C functions follow the pattern used in the C standard library, in other words, the caller must take care of the arguments given.

I hope with this explanation you could understand better how Netdata works, I do not wanna arrive here and say for you "Remove it because I want". I am giving this whole explanation for you understand the whole process and you arrive in the same conclusion.

Case you have any doubt about everything I wrote here, please, ask me that I will help you.

Best Regards!

Reply:
Hi @thiagoftsm,

thank you for explanation how netdata works. It is helpful! 

I have removed parameters checks to keep code aligned with current netdata design.

Thank you!

Reply:
I am glad to read this! Now you reached the level of understanding of the code that I wished when I said for you to read the files. :)

You have my promise that tomorrow, Sunday July 28th, I will test this PR.

You are welcome! And any doubt, let me know!

Reply:
Resolving this, it works fine, except for my comments.